packages/web3-utils/src/converters.ts

```diff
@@ -0,0 +1,340 @@
+	lovelace: expo10(6),
+	picoether: expo10(6),
+	gwei: expo10(9),
+	Gwei: expo10(9),
+	shannon: expo10(9),
+	nanoether: expo10(9),
+	nano: expo10(9),
+	szabo: expo10(12),
+	microether: expo10(12),
+	micro: expo10(12),
+	finney: expo10(15),
+	milliether: expo10(15),
+	milli: expo10(15),
+	ether: expo10(18),
+	kether: expo10(21),
+	grand: expo10(21),
+	mether: expo10(24),
+	gether: expo10(27),
+	tether: expo10(30),
+};
+
+const errorMessage = (value: unknown, error: string) =>
+	`Invalid value given "${String(value)}". Error: ${error}.`;
+
+export type EtherUnits = keyof typeof ethUnitMap;
+
+/** @internal */
+const bytesToBuffer = (data: Bytes): Buffer | never => {
+	if (Buffer.isBuffer(data)) {
+		return data;
+	}
+
+	if (data instanceof Uint8Array) {
+		return Buffer.from(data);
+	}
+
+	if (Array.isArray(data)) {
+		if (data.some(d => d < 0)) {
+			throw new Error(errorMessage(data, 'contains negative values'));
+		}
+
+		if (data.some(d => d > 255)) {
+			throw new Error(errorMessage(data, 'contains numbers greater than 255'));
+		}
+
+		if (data.some(d => !Number.isInteger(d))) {
+			throw new Error(errorMessage(data, 'contains invalid integer values'));
+		}
+
+		return Buffer.from(data);
+	}
+
+	if (typeof data === 'string' && !isHexStrict(data)) {
+		throw new Error(errorMessage(data, 'not valid hex string'));
+	}
+
+	if (typeof data === 'string' && (data.startsWith('0x') || data.startsWith('0X'))) {
+		return Buffer.from(data.substr(2), 'hex');
+	}
+
+	throw new Error(errorMessage(data, 'can not parse as byte data'));
+};
+
+/** @internal */
+const bufferToHexString = (data: Buffer) => `0x${data.toString('hex')}`;
+
+/**
+ * Convert a byte array to a hex string
+ */
+export const bytesToHex = (bytes: Bytes): HexString => bufferToHexString(bytesToBuffer(bytes));
+
+/**
+ * Convert a hex string to a byte array
+ */
+export const hexToBytes = (bytes: HexString): Buffer => bytesToBuffer(bytes);
+
+/**
+ * Converts value to it's number representation
+ */
+export const hexToNumber = (value: HexString): bigint | number => {
+	if (!isHexStrict(value)) {
+		throw new Error(errorMessage(value, 'not valid hex string'));
+	}
+
+	const [negative, hexValue] = value.startsWith('-') ? [true, value.substr(1)] : [false, value];
+	const num = BigInt(hexValue);
+
+	if (num > BigInt(Number.MAX_SAFE_INTEGER) || num < BigInt(Number.MIN_SAFE_INTEGER)) {
+		return negative ? BigInt(-1) * BigInt(num) : BigInt(num);
+	}
+
+	return negative ? -1 * Number(num) : Number(num);
+};
+
+/**
+ * Converts value to it's number representation @alias `hexToNumber`
+ */
+export const toDecimal = hexToNumber;
+
+/**
+ * Converts value to it's hex representation
+ */
+export const numberToHex = (value: Numbers): HexString => {
+	try {
+		if (typeof value === 'number' && !Number.isFinite(value)) {
+			throw new Error(errorMessage(value, 'contains invalid integer values'));
+		}
+
+		if ((typeof value === 'number' || typeof value === 'bigint') && BigInt(value) < BigInt(0)) {
```

Comment:
```suggestion
		if ((typeof value === 'number' || typeof value === 'bigint') && value < 0) {
```

Casting isn't necessary, if it's a number it's
`Number.MIN_SAFE_INTEGER <= value <= Number.MAX_SAFE_INTEGER` or else the user wouldn't be able to pass it

```typescript
-100 < 0 // true
BigInt(-100) < 0 // true
BigInt(Number.MIN_SAFE_INTEGER - 1) < 0 // true
```